core/dbt/events/format.py

```diff
@@ -3,6 +3,22 @@
 from typing import Optional, Union
 
 
+node_states = {
+    "queued": "queued",
+    "skipped": "skipped",
+    "started": "started",
+    "success": "success",
+    "failed": "failed"
+}
+
+node_statuses = {
+    "running": "running",
+    "pass": "pass",
+    "error": "error",
+    "fail": "fail"
+}
```

Comment:
It might feel a little boiler platey, but I think a better way to model this would be to have an ABC `NodeState` with subclasses for each of these. Those subclasses can override `__str__()` to get the string via something like `str(Queued())`. This way we can pass around `Queued()` and have mypy check it everywhere until the last minute where we need to turn it into a string.

Reply:
JK I think Jeremy had a better idea since it sounds like these are already defined somewhere else in core. Reusing those definitions, but keeping the string translations in here is probably a better way to go.